cleanlab_studio/studio/enrichment.py

```diff
@@ -0,0 +1,67 @@
+"""
+Methods for interfacing with Enrichment Projects.
+
+**This module is not meant to be imported and used directly.** Instead, use [`Studio.get_enrichment_project()`](../studio/#method-get_enrichment_project) to instantiate an [EnrichmentProject](#class-enrichmentproject) object.
+"""
+
+from __future__ import annotations
+from datetime import datetime
+from typing import Any, Dict, Optional
+
+from cleanlab_studio.internal.api import api
+
+
+def _response_timestamp_to_datetime(timestamp_string: str) -> datetime:
+    """
+    Converts the timestamp strings returned by the Cleanlab Studio API into datetime typing.
+    """
+    response_timestamp_format_str = "%a, %d %b %Y %H:%M:%S %Z"
+    return datetime.strptime(timestamp_string, response_timestamp_format_str)
+
+
+class EnrichmentProject:
+    """Represents an Enrichment Project instance, which is bound to a Cleanlab Studio account.
+
+    EnrichmentProjects should be instantiated using the [`Studio.get_enrichment_project()`](../studio/#method-get_enrichment_project) method.
+    """
+
+    def __init__(
+        self,
+        api_key: str,
+        id: str,
+        name: str,
+        target_column_in_dataset: str,
+        created_at_string: Optional[str] = None,
+    ) -> None:
+        """Initialize an EnrichmentProject.
+
+        **Objects of this class are not meant to be constructed directly.** Instead, use [`Studio.get_enrichment_project()`](../studio/#method-get_enrichment_project).
```

Comment:
it can have `to_json` and `from_json`. to the user will have a true project object to create with.

Reply:
1. Because this requires `api_key` I'm not sure we want to expose `from_json()` to the end user.
2. `to_json()` has been added.

Reply:
so the problem remains, user will use a dict instead of `EnrichmentProject`. 
is this a comment pattern in studio to require api_key explicitly. for cli like, aws, docker, they just read from a local file or ENV VAR. 
I would suggest to think about if we can do that.

Reply:
I strongly agree that this pattern is not ideal and would like to see it tackled in a major release increment.